Menu/MenuBuilder.php

```diff
@@ -70,77 +64,32 @@ public function __construct(Pool $pool, FactoryInterface $factory, MenuProviderI
     /**
      * Builds sidebar menu.
      *
-     * @return ItemInterface
+     * @param array $adminGroups
+     *
+     * @return \Knp\Menu\ItemInterface
      */
-    public function createSidebarMenu()
+    public function createSidebarMenu(array $adminGroups = array())
     {
-        $menu = $this->factory->createItem('root', array(
-            'extras' => array(
-                'request' => $this->request,
-            ),
-        ));
-
-        foreach ($this->pool->getAdminGroups() as $name => $group) {
```

Comment:
Can we stay BC here? What if we call ``createSidebarMenu`` the old way (without a parameter)?